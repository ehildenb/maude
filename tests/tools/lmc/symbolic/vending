#!/bin/sh

MAUDE_LIB=$srcdir/../../../src/Main
export MAUDE_LIB

../../../src/Main/maude \
  < $srcdir/vending.maude -no-banner -no-advise \
  > vending.out 2>&1

diff $srcdir/vending.expected vending.out > /dev/null 2>&1
