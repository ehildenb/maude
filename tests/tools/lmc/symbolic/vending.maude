load symbolic-checker

--- from the Maude manual 
--- http://maude.cs.uiuc.edu/maude2-manual/maude-manual.pdf

(mod NARROWING-VENDING-MACHINE is  
  sorts Coin Item Marking Money Conf .  
  subsort Coin < Money .  
  op __ : Money Money -> Money [assoc comm] .  
  subsort Money Item < Marking .  
  op __ : Marking Marking -> Marking [assoc comm] .  
  op <_> : Marking -> Conf .  
  op $ : -> Coin [format (r! o)] .  
  op q : -> Coin [format (r! o)] .  
  op a : -> Item [format (b! o)] .  
  op c : -> Item [format (b! o)] .  

  var M : Marking .  
  rl [buy-c] : < $ > => < c > .  
  rl [buy-c] : < M $ > => < M c > .  
  rl [buy-a] : < $ > => < a q > .  
  rl [buy-a] : < M $ > => < M a q > .  
  rl [change]: < q q q q > => < $ > .  
  rl [change]: < M q q q q > => < M $ > .  
endm)

--- narrowing-based logical search
(search [,4] < M:Money > ~>* < a c > .)  

