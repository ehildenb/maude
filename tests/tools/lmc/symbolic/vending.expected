Maude> 
	    Full Maude 2.7 June 18th 2014


	    Logical Model Checker January 10th 2012

Introduced module NARROWING-VENDING-MACHINE

search [,4] in NARROWING-VENDING-MACHINE : < M:Money > ~>* < a c > .

Solution 1
M:Money --> $ q q q

Solution 2
M:Money --> q q q q q q q

No more solutions.

Maude> Bye.
